### ISSUE TITLE
Rate calculation is inaccurate

### ISSUE THREAD

**Issue by Nubebuster** [NotEnoughUpdates/NotEnoughUpdates#730]
I am making this issue to document the problem described below. I am working on a fix myself.

## The problem
In this original issue from Jul 22, 2021, exactly 2 years ago, I describe the problem
Moulberry/NotEnoughUpdates#209

> the crops per minute counter is overall very inaccurate currently. I just did a farming contest and got 106k cocoa in 20 minutes. That comes out to be 5300 crops/m. Howerver, the tracker has shown about 4000-4500 mostly. Here and there for a split second it would max out on 5200. But in any case even the max that the overlay counter has shown was still less than the actual rate.
> https://www.youtube.com/watch?v=S_BATRwRY3g

>In the video I provided you can see that it's hard to determine which crop speed I am getting. The range is between 4800 and 5200 when I am consistently breaking cocoa. That's a pretty broad range.

**TL;DR** The rate calculation is inaccurate and not customizable
## The solution
>My suggestion is to add a slider configuration option to the farming skill overlay where you can change the timeframe that is used in calculating xp/hr, crops/m etc. This would also make me able to determine crop speed including time where I am switching cocoa lanes.

On top of the customization, a new system is needed as the current system is downright inaccurate.

## Progress
I will put tests to show how the old calculation of the skill overlay is inaccurate and, in my opinion, faulty.

**_22 June_** 
I have been working on the new system and I am currently making it ready for deployment. You can find the current progress in the branch on my repository [here](https://github.com/Nubebuster/NotEnoughUpdates/tree/skill-overlay)
At this moment I have added a crops/unit timer and I do not use the new calculation for the XP/h and Coins/h.

Currently I want feedback on whether this is a good change or not.
Once I am confident the new system is better than the old system, I will remove the old system to replace it. I will then also use the new system for the XP/h and Coins/h calculation.

## Tests
[Test 1](https://www.youtube.com/watch?v=UIwUkU0hKPc)
Timeframe config: 60s
Crops at 0:07 697343
Crops at 1:07 717273
Actual rate: 19930/m
Old calc: 20964 (5.2% off)
New calc: 19834 (-0.5% off)
Description: old system seems to be off even more at points during the video, the end value makes the old system look better than it actually was during the test. The new calc is increasingly accurate during the video. I will try to fix why it's not accurate at the start.

More test underway...

**Comment by NopoTheGamer** [comment]
@Nubebuster should this be closed now?
